packages/backend/src/apps/m365-excel/common/normalise-formula-input.ts

```diff
@@ -0,0 +1,73 @@
+import type { IJSONValue } from '@plumber/types'
+
+const DATA_EXFILTRATION_FORMULAS = [
+  'HYPERLINK',
+  'WEBSERVICE',
+  'IMPORTDATA',
+  'RTD',
+]
+
+const SYSTEM_COMMAND_EXECUTION_FORMULAS = ['CMD', 'EXEC', 'DDE', 'SHELL']
+
+const SENSITIVE_DATA_EXPOSURE_FORMULAS = [
+  'GETPIVOTDATA',
+  'INDIRECT',
+  'OFFSET',
+  'CELL',
+  'VLOOKUP',
+  'HLOOKUP',
+]
+
+// Define denylist that could be part of a malicious formula
+const BLACKLISTED_FORMULAS = [
+  ...DATA_EXFILTRATION_FORMULAS,
+  ...SYSTEM_COMMAND_EXECUTION_FORMULAS,
+  ...SENSITIVE_DATA_EXPOSURE_FORMULAS,
+]
+
+/**
+ * Based on the documentation of OWASP for CSV Injection
+ * https://owasp.org/www-community/attacks/CSV_Injection
+ * Note that excel formulas are case-insensitive
+ */
+export function sanitiseFormulaInput(input: string): string {
+  // Regex to match blacklisted formulas
+  const combinedRegex = new RegExp(`${BLACKLISTED_FORMULAS.join('|')}\\(`, 'gi')
```

Comment:
`combinedRegex` is static on every invocation of `sanitiseFormulaInput`, we can extract it out of the function

Reply:
ohya thanks i missed out on this